beacon-chain/state/state-native/getters_withdrawal.go

```diff
@@ -0,0 +1,49 @@
+package state_native
+
+import (
+	types "github.com/prysmaticlabs/prysm/consensus-types/primitives"
+	enginev1 "github.com/prysmaticlabs/prysm/proto/engine/v1"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/runtime/version"
+)
+
+// WithdrawalQueue returns the list of pending withdrawals.
+func (b *BeaconState) WithdrawalQueue() ([]*enginev1.Withdrawal, error) {
+	if b.version < version.Capella {
+		return nil, errNotSupported("WithdrawalQueue", b.version)
+	}
+
+	b.lock.RLock()
+	defer b.lock.RUnlock()
+
+	return b.withdrawalQueueVal(), nil
+}
+
+func (b *BeaconState) withdrawalQueueVal() []*enginev1.Withdrawal {
+	return ethpb.CopyWithdrawalSlice(b.withdrawalQueue)
+}
+
+// NextWithdrawalIndex returns the index that will be assigned to the next withdrawal.
+func (b *BeaconState) NextWithdrawalIndex() (uint64, error) {
+	if b.version < version.Capella {
+		return 0, errNotSupported("NextWithdrawalIndex", b.version)
+	}
+
+	b.lock.RLock()
```

Comment:
do we want a lock to read the version?

Reply:
Version? This is `NextWithdrawalIndex`. And we don't have a lock when reading the version.

Reply:
I think @potuz refers to `b.version < version.Capella`. This version field is something that could be updated while another thread could be trying to read it